mne/source_estimate.py

```diff
@@ -250,7 +250,11 @@ def read_source_estimate(fname, subject=None):
     """  # noqa: E501
     fname_arg = fname
     _validate_type(fname, 'path-like', 'fname')
-    fname = str(fname)
+
+    # Expand `~` if present
+    # We're not using _check_fname() here due to the particularities involved
+    # in STC file handling (-lh, -rh suffix) executed below
+    fname = op.expanduser(fname)
```

Comment:
Can't you still use `fname = _check_fname(fname, overwrite='read', must_exist=False)` despite these particularities?

Reply:
I had a look yesterday and felt that it would increase code complexity and reduce readability. I can have another look tomorrow with hopefully fresher eyes.

Reply:
Your proposal seems to do the trick. I've pushed this change.